pygqlc/GraphQLClient.py

```diff
@@ -758,10 +758,138 @@ def execute(self, query, variables=None):
     post_timeout = env.get('post_timeout', '60')
     if env_headers:
       headers.update(env_headers)
-    response = requests.post(env['url'], json=data, headers=headers, timeout=int(post_timeout))
+
+    # Use httpx with HTTP/2 support
+    with httpx.Client(http2=True) as client:
```

Comment:
I have found that in my local the problem of slowness still stands. A way to fix it would be to force IPv4 like this:
```suggestion
    with httpx.Client(http2=True, transport=httpx.HTTPTransport(local_address="0.0.0.0")) as client:
```
But I don't like this because it does not solve the actual problem, namely: *why is IPv6 slow on Linux clients?*

Reply:
@maneandrea 
This is an excellent question about networking that's directly relevant to your httpx implementation. The issue you're describing is quite common and has to do with how modern networks handle dual-stack IPv6/IPv4 connectivity.

When someone in Linux experiences slow requests that are fixed by forcing IPv4, these are the most likely reasons:

### 1. IPv6 Connection Timeouts

Many networks advertise IPv6 connectivity, but it might not be fully functional. When httpx (or any modern HTTP client) attempts to establish a connection:

- It performs DNS resolution which returns both IPv6 and IPv4 addresses
- By default, it attempts IPv6 connections first (modern preference)
- If the IPv6 connection attempt silently fails or times out, it falls back to IPv4
- This fallback process can take several seconds, leading to perceived slowness

### 2. Broken IPv6 Networks

Linux is particularly susceptible to this issue because:
- Most Linux distributions have IPv6 enabled by default
- Corporate networks, VPNs, or home routers may have misconfigured IPv6 settings
- The OS will attempt to use IPv6 but encounter routing problems

### 3. Happy Eyeballs Not Always Happy

Modern clients use an algorithm called "Happy Eyeballs" (RFC 8305) to try both IPv6 and IPv4 in a race condition, but:
- Not all implementations have optimal timeouts
- Some clients still prefer IPv6 even when it's consistently slower
- httpx's implementation might be working correctly, but OS-level networking could be the bottleneck

### Solution for Your Implementation

Given these issues, it would be helpful to add an option to force IPv4 in your GraphQLClient. You could modify the httpx client creation to support this:

```python
# Example of how to modify the execute method
with httpx.Client(http2=True, socket_options=[
    (socket.AF_INET6, socket.IPPROTO_IPV6, socket.IPV6_V6ONLY, 0)  # Prefer IPv4
]) as client:
    # ... existing code
```

Or even simpler, provide an `ipv4_only` parameter that users could set:

```python
# In __init__
self.ipv4_only = False

# In execute
with httpx.Client(http2=True, transport=httpx.HTTPTransport(
    local_address="0.0.0.0" if self.ipv4_only else None
)) as client:
    # ... existing code
```

This would give users an easy way to troubleshoot connectivity issues without having to modify the library code.

Would you like me to implement this improvement as part of the current PR? It would be a nice enhancement that addresses a real-world issue that some of your users might encounter.

Reply:
I think it's probably something with your network provider

Reply:
I've added a "patch" in this commit [c762364](https://github.com/valiot/pygqlc/pull/46/commits/c762364d9486eac5eeabbdc3767ab7b9e30a9cd1)

Reply:
Wouldn't it be better to pass it directly on the `.execute` method as well?

To give an example, `GraphQL` is a singleton so... if we have three files:
```
--- file1.py
    ----- Let the connection be decided here normally
--- file2.py
    ----- we want only IPv4
--- file3.py
    ----- same as file1.py
```

in the moment where we choose `pygqlc = GraphQLClient(ipv4_only=True)` on the file `file2.py`, we're going to change the entire behaviour of the other two files WITHOUT even knowing 👀 

i don't know if that's something that can cause errors on the future, it's only my mind looking for possible situations

Reply:
I think we do want the whole application to change to ipv4 if it's needed. If it is a problem for someone (@<!-- linear:userMention:a73d30e8-8336-4554-b0f0-0b10a6fa39b2 -->andrea.manenti right now) he would have to change all the places it's being called at. The server doesn't care if its ipv4 or 6 so the response will never change.